### PR TITLE
[compiler] IEmitCode = IEmitCodeGen[SValue]

### PR DESCRIPTION
**Pull request by patrick-schultz**
~Stacked on #10859~

### CODE REVIEW

**Comment by tpoterba** [comment]
Could you run benchmarks?

**Comment by patrick-schultz** [comment]
Yeah, I'm about to run benchmarks on the current top of the stack of SCode->SValue changes. Will post results here.

**Comment by patrick-schultz** [comment]
https://gist.github.com/patrick-schultz/8fcd19124266245ec456b5eefac5c516

File: hail/src/main/scala/is/hail/expr/ir/functions/UtilFunctions.scala [hunk 1]
```diff
@@ -306,63 +306,80 @@ object UtilFunctions extends RegistryFunctions {
...
+            {
+              r.toI(cb).consume(cb,
+                cb.assign(w, w | const(4)),
+                { b2 =>
+                  cb.assign(w, w | b2.asBoolean.boolCode(cb).mux(const(8), const(0)))
+                }
+              )
+            })
+
+          val Lpresent = CodeLabel()
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/expr/ir/functions/UtilFunctions.scala:379]: these are unused now, looks like

Comment (patrick-schultz) [reply hail/src/main/scala/is/hail/expr/ir/functions/UtilFunctions.scala]: Whoops, thanks

**Review by tpoterba — APPROVED** [review]

**Pull request merged**